### PR TITLE
Fix F12 goto in scripts #2850

### PR DESCRIPTION
**Pull request by majocha**
This is a fix for #2850

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs [hunk 1]
```diff
@@ -336,7 +336,10 @@ type internal FSharpGoToDefinitionService [<ImportingConstructor>]
                         return results.AsEnumerable()
                     else // we need to get an FSharpSymbol from the targetRange found in the signature
                          // that symbol will be used to find the destination in the corresponding implementation file
-                        let implFilePath = Path.ChangeExtension (sigDocument.FilePath,"fs")
+                        let implFilePath =
+                            // Bugfix: apparently sigDocument not always is a signature file
+                            if isSignatureFile sigDocument.FilePath then Path.ChangeExtension (sigDocument.FilePath, "fs") 
+                            else sigDocument.FilePath
```
Comment (cloudRoutine) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: Is this really the right place to do this? shouldn't it happen around ln 297?

Scripts probably need their own full branch of logic to handle [jumping across load directives](https://github.com/fsprojects/VisualFSharpPowerTools/blob/master/src/FSharp.Editing.VisualStudio/Navigation/GoToDefinitionFilter.fs#L72-L75) and making sure they end up at the right destination within a project (imho it should be in the impl file not the sig)

Comment (majocha) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: I admit all the branches are a bit overwhelming to me. I just identified the place where it breaks ATM. Full blown solution would be great :)

**Comment by vasily-kirichenko** [comment]
I pulled this fix into https://github.com/Microsoft/visualfsharp/pull/2855

**Comment by majocha** [comment]
Closing in favor of #2855

File: vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs (hunk 1 above)
Comment (majocha) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: I wonder if it wouldn't be a better solution to just setup script projects and compile script files just like the rest.
Now they are treated in a special and ephemeral way and that causes problems.

Comment (cloudRoutine) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: compile script files?

Comment (majocha) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: :) do whatever FSharpChecker now does to them but not necessarily wait for a file to be opened in a editor, which is the case now.

Comment (majocha) [inline vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs]: That's probably not that easy as it seems. I can imagine edge cases, but in a simple case of a script file inside a solution referencing other included scripts and projects I would expect it to just work. There shouldn't be any jumping through hoops to just get symbol info. Project system should handle this somehow, IMO.

**Pull request closed without merging**